### DISCUSSION TITLE
How would you replace all methods that have the name 'getSizeOfBasket'?

### DISCUSSION THREAD

**Discussion by christiancoleman** [vfsfitvnm/frida-il2cpp-bridge#441] [Q&A]
I'm playing with a game that has pretty consistent naming conventions for something I'd like to replace. It however is implemented in over 30 methods with varying names. I'm trying to do something like the following but it doesn't work for me. Any ideas or suggestions?

```
const snapshot = Il2Cpp.MemorySnapshot.capture();
snapshot.objects.forEach((instance: Il2Cpp.Object) => {
		if(instance?.method?.name == "GetBasketSize"
		 	|| instance?.method?.name == "GetSizeOfBasket" 
			|| instance?.method?.name == "GetBasketSizePlease"
			|| instance?.method?.name == "get_Basketsize"){
				const getBasketSizeGeneric = instance.method<number>(instance?.method?.name);
				getBasketSizeGeneric.implementation = function() : number{		
					return 5;
				}
			}
		if(instance?.class?.type?.name == 'WithBuddies.Common.ValidationResult') {
			console.log('~~~~~~~~~~~~~~~~~~~~~~~~VALIDATION RESULT FOUND');
			const get_IsValid = instance.method<boolean>("get_IsValid");
			get_IsValid.implementation = function() : boolean{				
				return true;
			}
		}
});
snapshot.free();
```

**Comment by DragonRiseo** [comment]
easy 
just make the if statment instead of the whole name of each one , just make it 
method.name.toLowerCase().include("basket" && "size") 
that is it , every method in that class that has both basket and size would be modified  

this would work for anything like : 
Get_BasKetSiZe  or baSket_Shiitless_size_hack_method or anything with basket and size

**Reply by christiancoleman** [reply]
Ok sweet. I was wondering about the logic surrounding it too. Is this a valid way of grabbing instance methods and then subsequently querying about their name and method? Just felt like it could be breaking somewhere since it didn't seem to hook anything when I tried it.

**Comment by DragonRiseo** [comment]
ya , just use the heap scanning , get your instance to the class , and then
hook anything you want in that class
